### PR TITLE
[IGA] Missing fixes

### PR DESCRIPTION
**Pull request by oberbichler**
today seems not to be my merge-day...

### CODE REVIEW

File: applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h [hunk 1]
```diff
@@ -93,37 +93,37 @@ class NodeCurveGeometry3D
...
         const VectorType& Value) override
     {
-        auto& node = *Node(Index);
- 
-        node[0] = Value[0];
-        node[1] = Value[1];
-        node[2] = Value[2];
+        Kratos::Node<3>& node = *Node(Index);
+
+        for (size_t i = 0; i < 3; i++) {
```
Comment (philbucher) [inline applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h]: Please use std::
(And ++i)

**Review by philbucher — COMMENTED** [review]
Mior comment, then I approve

File: applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 // Project includes
 #include "includes/define.h"
-#include "includes/Node.h"
+#include "includes/node.h"
```
Comment (loumalouomega) [inline applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h:23]: :D

File: applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h [hunk 3]
```diff
@@ -93,37 +93,37 @@ class NodeCurveGeometry3D
...
     void SetPole(
         const int Index,
         const VectorType& Value) override
     {
-        auto& node = *Node(Index);
- 
-        node[0] = Value[0];
-        node[1] = Value[1];
-        node[2] = Value[2];
+        Kratos::Node<3>& node = *Node(Index);
```
Comment (loumalouomega) [inline applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h]: Define typedef Node<3> NodeType;

Maybe in the future we remove the template from Node

File: applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h (hunk 1 above)
Comment (loumalouomega) [reply applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h]: Yes std, or you wil use c size_t, and you will have problems with pybind or something

File: applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h [hunk 4]
```diff
@@ -137,36 +137,36 @@ class NodeCurveGeometry3D
...
-     * \param Value New weight of the Kratos node
+     * @param Index Index of the node
+     * @param Value New weight of the Kratos node
      */
     void SetWeight(
         const int Index,
         const ScalarType Value) override
     {
-        auto& node = *Node(Index);
+        Kratos::Node<3>& node = *Node(Index);
```
Comment (loumalouomega) [inline applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h]: Idem, NodeType

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Almost, minor comments

File: applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h (hunk 1 above)
Comment (oberbichler) [reply applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h]: `++i`/`i++`: I dont see a difference
`size_t`: I forgot… Just to know: what is the exact problem?

File: applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h (hunk 3 above)
Comment (oberbichler) [reply applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h]: BTW: Does the current definition of the node template make sense? I think it always inherits from a Point<3> or something like that...

Comment (philbucher) [reply applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h]: Afaik also this template is gonna be removed

File: applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h (hunk 1 above)
Comment (philbucher) [reply applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h]: ++i is good practice, even if in this case it is not a difference

I don't know the exact problem with the pybind `size_t` but as far as I rememeber there were some problems in win

**Review by philbucher — APPROVED** [review]

**Review by loumalouomega — APPROVED** [review]
Ok

**Pull request merged**